relay/sources/relayd/src/output/upstream.rs

```diff
@@ -28,14 +28,46 @@
 // You should have received a copy of the GNU General Public License
 // along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
 
-use crate::configuration::UpstreamConfig;
-use reqwest::Client;
+use crate::Error;
+use crate::JobConfig;
+use futures::Future;
+use std::path::PathBuf;
+use std::sync::Arc;
+use tracing::{debug, span, Level};
 
-pub fn forward_file(cfg: UpstreamConfig) {
-    let client = Client::new();
-    let resp = client
-        .put("http://httpbin.org/")
-        .basic_auth("rudder", Some("rudder"))
-        .send()
-        .unwrap();
+pub fn send_report(
+    job_config: Arc<JobConfig>,
+    path: PathBuf,
+) -> Box<dyn Future<Item = (), Error = Error> + Send> {
+    let report_span = span!(Level::TRACE, "upstream");
+    let _report_enter = report_span.enter();
+    Box::new(forward_file(job_config, "report", path))
+}
+
+fn forward_file(
+    job_config: Arc<JobConfig>,
+    endpoint: &str,
+    path: PathBuf,
+) -> impl Future<Item = (), Error = Error> + '_ {
+    tokio::fs::read(path)
+        .map_err(|e| e.into())
+        .and_then(move |d| {
+            job_config
+                .clone()
+                .client
+                .clone()
```

Comment:
why the two clone ?

Reply:
the first is useless, good catch! (came from a refactoring a previous version)